### DISCUSSION TITLE
git diff --name-only and spaces

### DISCUSSION THREAD

**Discussion by eruffaldi** [community/community#22586] [New to GitHub]
<p>I had to recover the set of files modified during a specific commit as follows:</p>
<p>COMMIT=28cd71c36323b8ff4ff01cfabed0a1764d4744c3</p>
<p>git archive -o changed.zip $COMMIT $(git diff --name-only $COMMIT $COMMIT^)</p>
<p>If there are spaces in file names the above does not work.</p>
<p>The question is, how can deal correctly with spaces in names returned by git diff. There are several options involving sed/xargs, but I would like to have a feedback from you experts.</p>
<p>Thanks</p>

**Comment by eruffaldi** [comment, marked as answer]
<p>Apologies, it was easy:</p>
<p>git archive -o changed.zip $COMMIT $(git diff --name-only $COMMIT $COMMIT^ | sed ‘s/.*/"&amp;"/’)</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/eruffaldi">@eruffaldi</a>,</p>
<p>Thank you for sharing your solution! We’re looking forward to seeing you around!</p>
<p>Best,</p>
<p>Andrea</p>
